mde-services/src/main/java/de/terrestris/mde/mde_backend/controller/MetadataCollectionController.java

```diff
@@ -605,25 +608,43 @@ public ResponseEntity<Void> deleteComment(
             description =
                 "Not found: The provided metadata collection does not exist (or you don't have the permission to delete it)",
             content = @Content),
+        @ApiResponse(
+            responseCode = "409",
+            description =
+                "Conflict: The metadata collection is assigned to a user or you are not the owner or in the team and cannot be deleted",
+            content = @Content),
         @ApiResponse(
             responseCode = "500",
             description =
                 "Internal Server Error: Something internal went wrong while deleting the metadata collection",
             content = @Content)
       })
-  public ResponseEntity<?> delete(@PathVariable("metadataId") String metadataId) {
+  public ResponseEntity<?> delete(
+      @PathVariable("metadataId") String metadataId, Authentication auth) {
     try {
-      Optional<MetadataCollection> metadataCollection = service.findOneByMetadataId(metadataId);
-
-      if (metadataCollection.isEmpty()) {
-        log.warn("Could not find metadata collection with ID {}", metadataId);
-        return new ResponseEntity<>(NOT_FOUND);
+      MetadataCollection metadataCollection =
+          service
+              .findOneByMetadataId(metadataId)
+              .orElseThrow(() -> new ResponseStatusException(NOT_FOUND));
+      if (metadataCollection.getAssignedUserId() != null) {
+        return new ResponseEntity<>(CONFLICT);
+      }
+      if (!auth.getAuthorities().contains(new SimpleGrantedAuthority("MdeAdministrator"))) {
+        var userId = auth.getName();
+        if (metadataCollection.getOwnerId() == null
+            || metadataCollection.getTeamMemberIds() == null) {
+          return new ResponseEntity<>(CONFLICT);
+        }
+        if (!metadataCollection.getOwnerId().equals(userId)
+            && !metadataCollection.getTeamMemberIds().contains(userId)) {
+          return new ResponseEntity<>(CONFLICT);
+        }
```

Comment:
Absolutely non blocking, but should be include the checks in the service method? This way they would also be applied if it will be called from others.

Reply:
Yes, this should be refactored to a service method anyway since several deletions are taking place when including the GNOS records. Maybe we can do that when fixing the `TODO` regarding security